src/damoo.js

```diff
@@ -218,7 +243,11 @@
     };
 
     Thread.prototype.remove = function(d) {
-        var i = this.get(d).index;
+        var item = this.get(d);
+        if (item.link !== undefined) {
```

Comment:
I preferred `typeof val !== "undefined"` to detect undefined variables. :laughing: